genie-client/src/main/java/com/netflix/genie/client/interceptors/ResponseMappingInterceptor.java

```diff
@@ -55,6 +56,10 @@ public Response intercept(final Chain chain) throws IOException {
         if (response.isSuccessful()) {
             return response;
         } else {
+            if (response.code() == 429) {
+                throw new GenieClientTooManyRequestException();
```

Comment:
There should be an error message in the body you can parse out and use here instead of just bland. Can probably move this down into the below body parsing section and have the if somewhere else before final exception is thrown

Reply:
Done.